server/src/main/java/org/elasticsearch/common/io/DiskIoBufferPool.java

```diff
@@ -20,20 +21,33 @@
         ByteSizeValue.parseBytesSizeValue(System.getProperty("es.disk_io.direct.buffer.size", "64KB"), "es.disk_io.direct.buffer.size")
             .getBytes()
     );
-    public static final int HEAP_BUFFER_SIZE = 8 * 1024;
 
-    private static final ThreadLocal<ByteBuffer> ioBufferPool = ThreadLocal.withInitial(() -> {
+    // placeholder to cache the fact that a thread does not work with cached direct IO buffers in #ioBufferPool
+    private static final ByteBuffer EMPTY_BUFFER = ByteBuffer.allocate(0);
+
+    // protected for testing
+    protected static final ThreadLocal<ByteBuffer> ioBufferPool = ThreadLocal.withInitial(() -> {
```

Comment:
nit: rather than expose the thread local could we expose a protected method "clear()" that clears it instead?

Reply:
Since it's a nit I think, I'm gonna decline here I think :) Not sure it's worth adding extra lines of code for this. I have a follow-up queued here anyway that might make this class obsolete :D